enhancements/agent-installer/automated-workflow.md

```diff
@@ -0,0 +1,491 @@
+  future enhancement.
+- We are not currently considering replacing the IPI installer or
+  assisted installer SaaS.
+- Describing an end-to-end solution for users without their own
+  orchestration tool is left to a future enhancement.
+- Collecting telemetry data for connected deployments is left to a
+  future enhancement.
+- Details of securing the communication between agents and services to
+  ensure only desired hosts join the cluster are left to a future
+  enhancement.
+- Details of a workflow for deploying in settings with DHCP assignment
+  of IP addresses is left to a future enhancement.
+
+## Proposal
+
+We propose to extend the `openshift-install` binary with a new
+sub-command, `create image` for generating a bootable image containing
+all of the information needed to deploy a cluster.  A command line
+tool avoids issues with prerequisites for deploying a long-running
+service, and can be run on a low-resource host such as a laptop or
+even on a host at a different site before deployment begins.
+
+Most enterprise grade hardware includes a management controller that
+can present an ISO image to a host virtually, allowing the host to
+boot from the image remotely. Hypervisors such as kvm and vSphere also
+support booting from ISOs. Therefore, we will concentrate on that
+format and approach first. Building minimal ISOs with separate root
+filesystems or images that can be used with hosts that PXE boot will
+come later.
+
+We propose to leave up to the user's orchestration tool the
+responsibilities of managing and storing the images, serving them on
+the network, and especially configuring hosts to boot them. This gives
+users the flexibility to perform those operations in any way they
+like, with whatever tool they prefer. A design for an all encompassing
+approach based on tools provided entirely by Red Hat is left for a
+later enhancement.
+
+The `create image` command will need to take inputs similar to what
+the baremetal IPI installer or ZTP APIs use to describe the cluster
+(name, API IPs, etc.) and host-specific network settings (static IPs,
+VLANs, etc.). Using the existing file and custom resource types allows
+us to use the same version management techniques that we do for the
+APIs within a cluster.
+
+The `create image` command will produce a single image as output, to
+save storage space and reduce management complexity. The assisted
+installer service can already generate images with boot scripts that
+recognize which host is booting and apply host-specific network
+settings and roles (such as running different services) accordingly,
+and we will reuse that technique (and code).
+
+One host, designated as "node0", will be configured to run the
+assisted service and a client that uses the assisted service API to
+automate the deployment.
+
+All other hosts will run the assisted agent, configured to talk to the
+service on node0.
+
+When all of the images are booted, the client on node0 will use the
```

Comment:
I'm assuming that (1) the input file lists the hosts and (2) the client is polling the inventory via REST API until all agents check in?